src/cart/src/cart/crt_init.c

```diff
@@ -158,13 +158,6 @@ static int data_init(crt_init_options_t *opt)
 		D_WARN("CRT_CTX_NUM has no effect because CRT_CTX_SHARE_ADDR "
 		       "is not set or set to 0\n");
 
-	if (opt) {
-		if (opt->cio_fault_inject)
-			d_fault_inject_enable();
-		else
-			d_fault_inject_disable();
-	}
```

Comment:
Not sure why this is removed. Even if it is missed initially when it is not set, removing this part will not enable fault injection as it is disabled by default. This however prevents app that enables it from crt_init_opt() to function as before.

If we dont want for some reason to have this ability via crt_init_opt() then opt->cio_fault_inject should also be removed from the corresponding structure.

However, i still dont quite understand why we want to remove this

Reply:
A few notes here

1. in daos_crt_init_opt_get(), there is strange logic which makes opt to be NULL on the server side for some cases, then this opt for debugging will never work.
2. on client side, it seems there is no position to set cio_fault_inject at all, so it will always disable the inject_fault, unfortunately after it analyze the inject_fault yml file, which makes the inject never work on the client side.
3. I had thought to remove cio_fault_inject, but I found sth in crt_hdlr_ctl_fi_toggle(), not sure what it does. Do you think I can remove this too?

So I just remove this part, and only setting this if there is fault_inject.yml is provided. which seems reasonable to me.

Reply:
I thought at least part of the reason here was that (for the server at least) there was a requirement that it should explicitly enable FI at startup, to prevent a denial of service attack at run-time.  This check would have meant that you could enable the code, but not enable any fault locations, if the code is not enabled at start then it is not possible to enable it later on.

Reply:
I thought this really depends on the requirement of fault injection.  Why do we need enable this during the run time?  Since this is only needed for developing/debugging purpose, it looks like only enabling this during server startup is good enough?

Reply:
The idea is that some tests will want to enable this during runtime for a small portion of a test. A test might want to setup everything properly first, and then when ready in a controlled manner enable fault injection, do the acitons and disable fi again

Reply:
I think fault_inject itself, described by yml, already support enable/disable the fault injection.  so we probably do no need extra mechanism here?

Reply:
This probably should be taken up with daos validation who will be main users of this feature. We added runtime enable/disable cause we thought it would be handy, but if daos validation is fine with just altering individual faults at runtime that works as well. I dont have strong opinion on this either way

Reply:
I will remove this for now for simplicity. and I can add it back if we really need it.